genai-perf/genai_perf/parser.py

```diff
@@ -246,6 +246,49 @@ def _check_load_manager_args(args: argparse.Namespace) -> argparse.Namespace:
     return args
 
 
+def is_valid_url(url):
+    """
+    Checks if a URL is valid. It must use 'http' or 'https', have a valid
+    netloc, optional port, and contain '/metrics' in the path.
+    """
+    try:
+        result = urlparse(url)
+
+        # Check scheme
+        if result.scheme not in ["http", "https"]:
+            return False
+
+        # Check netloc
+        if not result.netloc:
+            return False
+
+        # Check path contains '/metrics'
+        if "/metrics" not in result.path:
+            return False
+
+        return True
+    except Exception:
+        return False
+
+
+def _check_server_metrics_url(
+    parser: argparse.ArgumentParser, args: argparse.Namespace
+) -> argparse.Namespace:
+    """
+    Checks if the server metrics URL is valid based on the backend and service kind.
+    """
+
+    # Check if the URL is valid and contains the expected path
+    if (
+        args.service_kind == "triton"
+        and args.server_metrics_url
+        and not is_valid_url(args.server_metrics_url)
+    ):
+        parser.error("The --server-metrics-url option contains an invalid URL format.")
```

Comment:
I would move this exception into is_valid_url.
And update this method to check service kind and existence of the metrics url. Then return the result of _is_valid_url. 

This way, as we add more checks later, we can make the call to the url validation method and it will bubble up the exception if one is thrown.